airflow-core/src/airflow/models/dagbag.py

```diff
@@ -470,9 +470,10 @@ def _load_modules_from_zip(self, filepath, safe_mode):
 
     def _process_modules(self, filepath, mods, file_last_changed_on_disk):
         from airflow.models.dag import DAG  # Avoid circular import
+        from airflow.sdk import DAG as SDKDAG
         from airflow.sdk.definitions._internal.contextmanager import DagContext
 
-        top_level_dags = {(o, m) for m in mods for o in m.__dict__.values() if isinstance(o, DAG)}
+        top_level_dags = {(o, m) for m in mods for o in m.__dict__.values() if isinstance(o, (DAG, SDKDAG))}
```

Comment:
```suggestion
        top_level_dags = {(o, m) for m in mods for o in m.__dict__.values() if isinstance(o, DAG)}
```

Reply:
We should update the type hint on L504 too

Reply:
or update 

https://github.com/apache/airflow/blob/fc63fb1d8a4d9851d5b8424859d8055dbcc25632/airflow-core/src/airflow/models/dagbag.py#L70